service/history/workflowcache/cache.go

```diff
@@ -133,16 +133,29 @@ func (c *wfCache) allow(domainID string, workflowID string, rateLimitType rateLi
 
 	switch rateLimitType {
 	case external:
-		return value.externalRateLimiter.Allow()
+		if !value.externalRateLimiter.Allow() {
```

Comment:
since I'm not familiar with these two limits in history: what's the difference?  and is it worth separating them in metrics / likely worth adding to logs?

Reply:
One is for ratelimiting external calls, one for internal (like the domain global limits) - I have seperated out the metrics and added a tag to the log - please have a look?